packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,697 @@
+const mockedGetAnonId = jest.fn();
+const mockedLogError = jest.fn();
+const createMockedConfig = ( isDevelopmentMode = false ): Config => ( {
+	logError: mockedLogError,
+	fetchExperimentAssignment: mockedFetchExperimentAssignment,
+	getAnonId: mockedGetAnonId,
+	isDevelopmentMode,
+} );
+const DEVELOPMENT_MODE = true;
+
+function mockFetchExperimentAssignmentToMatchExperimentAssignment(
+	experimentAssignment: ExperimentAssignment
+) {
+	spiedMonotonicNow.mockImplementationOnce( () => experimentAssignment.retrievedTimestamp );
+	mockedFetchExperimentAssignment.mockImplementationOnce( () =>
+		delayedValue(
+			{
+				ttl: experimentAssignment.ttl,
+				variations: {
+					[ experimentAssignment.experimentName ]: experimentAssignment.variationName,
+				},
+			},
+			ONE_DELAY
+		)
+	);
+}
+
+function setBrowserContext() {
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore
+	global.window = {};
+}
+
+function setSsrContext() {
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore
+	global.window = undefined;
+}
+
+beforeEach( () => {
+	jest.resetAllMocks();
+	setBrowserContext();
+} );
+
+describe( 'createExPlatClient', () => {
+	it( `should throw if initialized outside of a browser context`, () => {
+		setSsrContext();
+
+		expect( () => createExPlatClient( createMockedConfig() ) ).toThrowErrorMatchingInlineSnapshot(
+			`"Running outside of a browser context."`
+		);
+	} );
+
+	it( `shouldn't call any deps if only initialized`, () => {
+		createExPlatClient( createMockedConfig() );
+
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+
+	it( `shouldn't throw if initialized multiple times`, () => {
+		const a = createExPlatClient( createMockedConfig() );
+		const b = createExPlatClient( createMockedConfig() );
+		const c = createExPlatClient( createMockedConfig() );
+		const d = createExPlatClient( createMockedConfig() );
+		expect( a ).not.toBe( null );
+		expect( b ).not.toBe( null );
+		expect( c ).not.toBe( null );
+		expect( d ).not.toBe( null );
+	} );
+} );
+
+describe( 'ExPlatClient.loadExperimentAssignment single-use', () => {
+	it( `should successfully load an ExperimentAssignment`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig() );
+		spiedMonotonicNow.mockImplementationOnce(
+			() => validExperimentAssignment.retrievedTimestamp + 1000
+		);
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( validExperimentAssignment );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+	it( `[anonId] should successfully load an ExperimentAssignment`, async () => {
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( 'the-anon-id-123', ZERO_DELAY ) );
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
```

Comment:
For this test to match our expectations, we should set up the mock so that it only returns the experiment assignment if the anon ID is `the-anon-id-123`.

If we make the mocking changes suggested above, it'd be something like `mockFetchExperimentAssignmentToMatchExperimentAssignment( { experimentAssignment: validExperimentAssignment, expectedAnonId: 'the-anon-id-123' } )`